Applications/CLI/NumericsConfig.h

```diff
@@ -52,13 +61,37 @@ namespace detail
 
 #endif    // USE_LIS
 
-
 //
 // Global executor and vector/matrix builder.
 //
 #include "AssemblerLib/SerialExecutor.h"
-#include "AssemblerLib/SerialVectorMatrixBuilder.h"
+#include "AssemblerLib/GlobalSetup.h"
+
+#ifdef USE_PETSC
+    #include "AssemblerLib/PETScVectorMatrixBuilder.h"
+    #include "MathLib/LinAlg/PETSc/PETScVector.h"
+    #include "MathLib/LinAlg/PETSc/PETScMatrix.h"
+    #include "MathLib/LinAlg/PETSc/PETScLinearSolver.h"
+namespace detail
+{
+using GlobalExecutorType = AssemblerLib::SerialExecutor;
+
+using GlobalVectorMatrixBuilderType =
+        AssemblerLib::PETScVectorMatrixBuilder<
+            MathLib::PETScMatrix, MathLib::PETScVector>;
+}
 
+///
+/// Global setup collects the previous configuration in single place.
+///
+using GlobalSetupType =
```

Comment:
GlobalSetupType can be shared with non-PETSc case, because  detail::LinearSolverType points to PETSc solver